src/EventStore.Projections.Core/Services/Processing/ResultEventEmitter.cs

```diff
@@ -16,6 +18,16 @@ public class ResultEventEmitter : IResultEventEmitter {
 			return CreateResultUpdatedEvents(partition, result, at);
 		}
 
+		private void CheckStateSize(EmittedEventEnvelope result) {
+			if (System.Text.ASCIIEncoding.Unicode.GetByteCount(result.Event.Data) >= 8_000_000 && !_largeStateWarningLogged) {
```

Comment:
arguments are evaluated from left to right, so it would make sense to check the bool first since it's quicker. counting the bytes is quite expensive. it could even be an early return